openshiftcontrolplane/v1/types.go

```diff
@@ -142,6 +157,18 @@ type ImagePolicyConfig struct {
 	// additionalTrustedCA is a path to a pem bundle file containing additional CAs that
 	// should be trusted during imagestream import.
 	AdditionalTrustedCA string `json:"additionalTrustedCA"`
+
+	// imageStreamImportMode provides the import mode value for  imagestreams.
+	// It can be `Legacy` or `PreserveOriginal`. `Legacy` indicates that the legacy behaviour
+	// should be used. For manifest lists, the legacy behaviour will discard the manifest list
+	// and import a single sub-manifest. In this case, the platform is chosen in the following
+	// order of priority: 1. tag annotations; 2. control plane arch/os; 3. linux/amd64; 4. the first
+	// manifest in the list. `PreserveOriginal` indicates that the original manifest will be preserved.
+	// For manifest lists, the manifest list and all its sub-manifests will be imported.If this value
+	// is specified, this setting is applied to all newly created imagestreams which do not have the
+	// value set.
+	// +optional
+	ImageStreamImportMode ImportModeType `json:"imageStreamImportMode"`
```

Comment:
Can't see any tests for this field. Also, should this field also be gated behind the featuregate?

Reply:
@JoelSpeed apologies again for my limited knowledge on this - this is not an API that can be directly controlled by the user, but it still needs to be behind a feature gate. This value is set by the openshift-apiserver once it reads the image config status.

Reply:
If you don't have it gated, you've effectively GA'd this straight away, at which point we can't remove it. Since the other field is gated, I'd expect them both to be gated, and to have them promoted at the same time

Reply:
> Can't see any tests for this field

afaik the fields here are synced by the openshift-apiserver-operator in the observed config from the value of the status fields in the various other config CRs. This value is meant to be generated the same way - by looking at the `image.config.openshift.io` status and putting that in the observedconfig.

Reply:
@JoelSpeed about my above question - are tests needed for this? i believe this is the only open item left to resolve.

Reply:
The `OpenShiftAPIServer` api is where this value is reflected in the observed config. There is a basic test for that API [here](https://github.com/openshift/api/blob/0d46442e8df17a87cfcf6666ab19b28e88620b59/operator/v1/tests/openshiftapiservers.operator.openshift.io/AAA_ungated.yaml). I am not sure how we can add a test for an object which is generated by the apiserver.

Reply:
I also think the `OpenshiftAPIServerConfig` was used in v3.11, but was deprecated since.

Reply:
There's an explanation of how those style of tests work in https://github.com/openshift/api/tree/master/tests, but yes, I would expect some basic testing even now.

The openshift-apiserver isn't the only implementation of these APIs, other topologies of OpenShift use the CRDs, so we should make sure the validations work on the CRDs too

Reply:
thanks for the pointers. added a test for the featuregate.

Reply:
@JoelSpeed is the test file that i added sufficient?